codegen/method.go

```diff
@@ -1013,11 +1008,6 @@ func (ms *MethodSpec) setParseQueryParamStatements(
 
 		identifierName := camelCase(longQueryName) + "Query"
 
-		httpRefAnnotation := field.Annotations[antHTTPRef]
```

Comment:
this is not correct if the field.Annotation is for header then we cannot read from query params and must read from the header.

Same if the antHTTPRef annotation is body

Reply:
Done